src/pause.sol

```diff
@@ -19,17 +19,21 @@ import {DSNote} from "ds-note/note.sol";
 import {DSAuth, DSAuthority} from "ds-auth/auth.sol";
 
 contract DSPause is DSAuth, DSNote {
-    // --- auth ---
-    function setOwner(address owner_) public {
-        require(msg.sender == address(proxy), "ds-pause-undelayed-ownership-change");
+    // --- admin ---
+    modifier wait() {
+        require(msg.sender == address(proxy), "ds-pause-undelayed-call"); _;
+    }
+    function setOwner(address owner_) public wait {
         owner = owner_;
         emit LogSetOwner(owner);
     }
-    function setAuthority(DSAuthority authority_) public {
-        require(msg.sender == address(proxy), "ds-pause-undelayed-authority-change");
+    function setAuthority(DSAuthority authority_) public wait {
         authority = authority_;
         emit LogSetAuthority(address(authority));
     }
+    function setDelay(uint delay_) public note wait {
```

Comment:
Have you considered making functions that are not called internally `external`?

Reply:
> Have you considered making functions that are not called internally `external`?

👍 for this

Reply:
Reason I bring it up is because Lev a few days ago mentioned how solc goes through some hoops to make the function available both as a jump destination and as a callable function, which I imagine would make FV/bytecode audits harder/more cumbersome.

I don't have more details but would be very interested to have a look if this is documented somewhere.

Reply:
This is a good idea. Will address in a separate PR.